docs/assemblies/assembly-multi-tenancy-filter.adoc

```diff
@@ -0,0 +1,27 @@
+// file included in the following:
+//
+// assembly-built-in-filters.adoc
+
+[id='assembly-multi-tenancy-filter-{context}']
+= (Preview) Multi-tenancy filter
+
+[role="_abstract"]
+Kroxylicious’s Multi-tenancy filter presents a single Kafka cluster to tenants as if it were multiple clusters.
+Tenant-specific operations are isolated by prefixing resources with an identifier.
+
+NOTE: This filter is currently in incubation and available as a preview. 
+We would not recommend using it in a production environment.
+
+The Multi-tenancy filter works by intercepting all Kafka RPCs (remote procedure calls) that reference resources, such as topic names and consumer group names:
+
+Request path:: On the request path, resource names are prefixed with a tenant identifier.
+Response path:: On the response path, the prefix is removed.
+
+Kafka RPCs that list resources are filtered so that only resources belonging to the tenant are returned, effectively creating a private cluster experience for each tenant.
+
```

Comment:
Should we call out that users are shared between **all** tenants? We don't intercept authentication and apply tenant prefixes or amend ACL's?

This is as much a question for @k-wall and the rest of @kroxylicious/developers as @PaulRMellor ...

Reply:
There's loads of gaps (functional and non-functional) because we've stopped working on it.     How would we decide what to call out in the docs?  Also we have a warning above advising this filter is incubating, so maybe that's enough.

I think it would be a better use of time to capture the gaps as Github issues and just link to the github label query from the docs.  

https://github.com/kroxylicious/kroxylicious/issues/272

Reply:
Yeah there a gaps but the wording of this section implies a fairly comprehensive level of isolation, and the one critical gap for me is that identity it not scoped. So I think that is worth calling out directly. Its a gap we already know about, I'm not calling for an audit of what it does and doesn't do. To me incubating implies there might the odd RPC missed or we haven't focused on its performance - i.e. its not a complete or rounded solution, but known gaps are reasonable to call out

Reply:
Thanks @SamBarker and @k-wall.  Trying to address both your concerns, I added the following note with a link to github issues:

```
IMPORTANT: While the Multi-tenancy filter isolates operations on resources, it does not isolate user identities across tenants. 
User authentication and ACLs (Access Control Lists) are shared across all tenants, meaning that identity is not scoped to individual tenants. 
For more information on open issues related to this filter, see {github-issues}.
```